core/unit_test/TestViewMapping_a.hpp

```diff
@@ -667,45 +667,45 @@ void test_view_mapping() {
     ASSERT_EQ(vr1.data(), &data[0]);
     ASSERT_EQ(cr1.data(), &data[0]);
 
-    ASSERT_TRUE((std::is_same_v<typename T::data_type, int*>));
-    ASSERT_TRUE((std::is_same_v<typename T::const_data_type, const int*>));
-    ASSERT_TRUE((std::is_same_v<typename T::non_const_data_type, int*>));
+    static_assert(std::is_same_v<typename T::data_type, int*>);
```

Comment:
Some what of a drive-by change but we are now guarding the test more restrictively at runtime the checks here need to be compile-time to still work despite the guard.